deployment/README.md

```diff
@@ -27,51 +100,27 @@ and then exposed for use in persistent environments like testnet/mainnet.
 contracts (like MCMS, LinkToken etc) which can be shared
 by products.
 
-/deployment/ccip
-- package name `ccipdeployment`
-- Files and tests per product deployment/configuration workflows
-- Tests can use deployment/memory for fast integration testing
-- TODO: System state representation is defined here, need to define
-  an interface to comply with for all products.
-
-/deployment/ccip/changeset
-- package name `changeset` imported as `ccipchangesets`
-- These function like scripts describing state transitions
-  you wish to apply to _persistent_ environments like testnet/mainnet
-- They should be go functions where the first argument is an
-  environment and the second argument is a config struct which can be unique to the 
-  changeset. The return value should be a `deployment.ChangesetOutput` and an error.
-
--  `do_something.go`
-    ```Go
-    func DoSomethingChangeSet(env deployment.Environment, c ccipdeployment.Config) (deployment.ChangesetOutput, error)
-    {
-        // Deploy contracts, generate MCMS proposals, generate
-        // job specs according to contracts etc.
-        return deployment.ChangesetOutput{}, nil
-    }
-    ```
-
--  `do_something_test.go`
-    ```Go
-    func TestDoSomething(t *testing.T)
-    {
-        // Set up memory env
-        // DoSomethingChangeSet function
-        // Take the artifacts from ChangeSet output
-        // Apply them to the memory env
-        // Send traffic, run assertions etc.
-    }
-    ```
-- Changesets are exposed and applied via a different repo. 
-
-/deployment/llo
-- package name `llodeployment`
-- Similar to /deploymet/ccip, these are product-specific deployment/configuration workflows
-- Tests can use deployment/memory for fast integration testing
-
-/deployment/llo/changeset
-- package name `changeset` imported as `llochangesets`
-- Similar to deployment/ccip/changesets
-- These function like scripts describing state transitions
-  you wish to apply to _persistent_ environments like testnet/mainnet
+/deployment/<product>
+- package name `<product>deployment`
+- Internal building blocks for changesets  
+- TODO: can we make this `internal`?
+
+/deployment/<product>/changeset
+- Think of this as the public API for deployment and configuration
+of your product. 
+- All the changesets should have an associated test using a memory or devenv
+environment.
+- package name `changeset` imported as `<package>changeset`
+ 
+## Integration testing
+Integration tests should live in the integration-tests/go.mod module and leverage
+the deployment module for product deployment and configuration. The integration tests
+should only depend on deployment/<product>/changeset and deployment/environment. 
+
+## FAQ
+### Should my changeset be idempotent? 
```

Comment:
Not relevant for this documentation change, but it again brings up a question of whether we want to make idempotency an explicit signal - that is, if you are making an idempotent change, it should have some explicit signal that it is. Probably too soon to decide this, but it's on my mind.

Reply:
I still kinda think it's ok to leave up to product teams as long as we have some examples of both